dynamics/apis/api_base.py

```diff
@@ -277,6 +277,13 @@ def _bulk_post_request(self, data, isolation: str):
 
         if response.status_code == 200 or response.status_code == 201:
             result = json.loads(response.text)
+
+            error_messages = [resp.get("body", {}).get("error", {}).get("message", None) for resp in result.get("responses", []) if resp.get("status", None) == 400]
+            error_messages = [message for message in error_messages if message is not None]
+
+            if error_messages:
+                if purchase_invoice_id: self._delete_request({}, '/purchaseInvoices({0})'.format(purchase_invoice_id))
+                raise WrongParamsError(error_messages)
```

Comment:
can do something like this

message and response seperatly
raise WrongParamsError(message='some string', response=error_messages)